mypy/checker.py

```diff
@@ -2336,6 +2336,28 @@ def check_assignment(self, lvalue: Lvalue, rvalue: Expression, infer_lvalue_type
                 self.infer_variable_type(inferred, lvalue, rvalue_type, rvalue)
             self.check_assignment_to_slots(lvalue)
 
+    def check_except_reachability(self, s: TryStmt) -> None:
+        """Perform except block reachability checks.
+
+        1. Check for duplicate exception clauses.
+        2. Check if super class has already been caught.
+        """
+        seen: List[Type] = []
+        for expr in s.types:
+            if expr and isinstance(expr, NameExpr) and isinstance(expr.node, TypeInfo):
+                with self.expr_checker.msg.disable_errors():
+                    typ = get_proper_type(self.check_except_handler_test(expr))
+                if isinstance(typ, AnyType):
```

Comment:
This can be a `Union` with `Any` type as well 🤔

Reply:
Can you think of a case where this happens?

I would argue that his can only happen for variables and tuples (which are both filtered with the `if expr and isinstance(expr, NameExpr) and isinstance(expr.node, TypeInfo):`)

For safety reasons: I can add Union to the check `if isinstance(typ, AnyType) or isinstance(typ, UnionType):
`